README.md

```diff
@@ -1,16 +1,32 @@
+## Pre-requisites
+
+- You will need to have an Azure account already available
+- Azure CLI installed to use this provider.
+- An Azure OpenAI endpoint provisioned with the models you would like to use.
+
+You will need to provide the following information when first setting up the provider:
+
+- Azure Subscription ID
+- Azure Resource Group
+- Azure OpenAI model name
+
 ## Usage Example
 
+To override the default model of all your gptscripts you can run the following command:
+
 ```
 gptscript --default-model='gpt-4o from github.com/gptscript-ai/azure-openai-provider' examples/helloworld.gpt
 ```
 
+### Troubleshooting
 
-#### Advanced
+#### I'm getting 429 errors due to rate limiting
 
-If you want to bypass the need for the Azure CLI or authentication to Azure in general, you can set `GPTSCRIPT_AZURE_ENDPOINT`, `GPTSCRIPT_AZURE_API_KEY`, `GPTSCRIPT_AZURE_DEPLOYMENT_NAME` as environment variables before using the provider for the first time. Note that these values will still be saved as a GPTScript credential, so changing the environment variables later will not effect GPTScript until you delete that saved credential.
+ You need to increase the rate limit of your Azure OpenAI deployment [quota](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/quota?tabs=rest).
 
+### Development
 
-## Development
+#### Advanced
```

Comment:
What does Advanced mean in this context since you deleted the advanced usage stuff I had before? All that is in here now is just regular development info.